src/Microsoft.TemplateSearch.TemplateDiscovery/PackChecking/PackChecker.cs

```diff
@@ -0,0 +1,133 @@
+                    ProduceAdditionalDataForPack(additionalDataProducers, checkResult, environment);
+                }
+                else
+                {
+                    IReadOnlyList<ITemplateInfo> foundTemplates = new List<ITemplateInfo>();
+                    checkResult = new PackCheckResult(packInfo, foundTemplates);
+                }
+            }
+            catch // (Exception ex)
+            {
+                // TODO: abstract the logging away from the console.
+                //Console.WriteLine($"Error attempting to install template pack {packInfo.Id}.");
+                //Console.WriteLine(ex.Message);
+                IReadOnlyList<ITemplateInfo> foundTemplates = new List<ITemplateInfo>();
+                checkResult = new PackCheckResult(packInfo, foundTemplates);
+            }
+
+            if (!persistHive)
+            {
+                TryCleanup(environment);
+            }
+
+            return checkResult;
+        }
+
+        private void ProduceAdditionalDataForPack(IReadOnlyList<IAdditionalDataProducer> additionalDataProducers, PackCheckResult packCheckResult, EngineEnvironmentSettings environment)
+        {
+            if (!packCheckResult.AnyTemplates)
+            {
+                return;
+            }
+
+            foreach (IAdditionalDataProducer dataProducer in additionalDataProducers)
+            {
+                dataProducer.CreateDataForTemplatePack(packCheckResult.PackInfo, packCheckResult.FoundTemplates, environment);
+            }
+        }
+
+        private bool TryInstallPackage(string packageFile, EngineEnvironmentSettings environment, out IReadOnlyList<ITemplateInfo> installedTemplates)
+        {
+            ((SettingsLoader)(environment.SettingsLoader)).UserTemplateCache.Scan(packageFile);
+            environment.SettingsLoader.Save();
+
+            if (((SettingsLoader)environment.SettingsLoader).UserTemplateCache.TemplateInfo.Count > 0)
+            {
+                installedTemplates = ((SettingsLoader)environment.SettingsLoader).UserTemplateCache.TemplateInfo;
+            }
+            else
+            {
+                installedTemplates = new List<ITemplateInfo>();
+            }
+
+            return installedTemplates.Count > 0;
+        }
+
+        private static ITemplateEngineHost CreateHost(IPackInfo packInfo)
+        {
+            string hostIdentifier = HostIdentifierBase + packInfo.Id;
+
+            ITemplateEngineHost host = TemplateEngineHostHelper.CreateHost(hostIdentifier);
+
+            return host;
+        }
+
+        private void TryCleanup(EngineEnvironmentSettings environment)
+        {
+            Paths paths = new Paths(environment);
+
+            try
+            {
+                paths.Delete(paths.User.BaseDir);
+            }
+            catch // (Exception ex)
+            {
+                //Console.WriteLine($"Error deleting BaseDir = {paths.User.BaseDir} under the temporary hive. Error: {ex.Message}");
```

Comment:
Nit: Can probably remove these or put behind a Debugger.IsAttached or something

Reply:
Removed. They're leftover from the initial round of work.